framework/rpc/grpc_testing.py

```diff
@@ -159,6 +161,25 @@ def set_not_serving(self):
             rpc="SetNotServing", req=empty_pb2.Empty(), log_level=logging.INFO
         )
 
+    def send_hook_request_start_server(self):
```

Comment:
https://github.com/grpc/grpc/pull/34519#discussion_r1340625700:

`HookService` [is not a part of](https://github.com/grpc/grpc/blob/e76ac4ab07d936a857339fb1bd4eeb621bd3fba6/src/proto/grpc/testing/test.proto#L95-L110) `XdsUpdateHealthService`, so it shouldn't be in `XdsUpdateHealthServiceClient`.

We should create another class called `HookServiceClient`.

Reply:
[`SendHookRequest` _is_ and that's what's being invoked here](https://github.com/grpc/grpc/blob/a17f08b49d7e46a2a102c84f57086fd1e6312113/src/proto/grpc/testing/test.proto#L109). There are two different RPCs here:

- `HookService.Hook`, which you pointed out here and which is called by `grpcurl`
- `XdsUpdateHelathService.SendHookRequest`, which controls a server running `HookService.Hook` and which is the actual RPC method referenced here.

Reply:
Oh, I see. Makes sense now, I missed this.